src/main/java/com/graphhopper/converter/resources/ConverterResourceGisgraphy.java

```diff
@@ -0,0 +1,188 @@
+package com.graphhopper.converter.resources;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import com.codahale.metrics.annotation.Timed;
+import com.graphhopper.converter.api.GisgraphyGeocodingResult;
+import com.graphhopper.converter.api.GisgraphySearchResult;
+import com.graphhopper.converter.api.Status;
+import com.graphhopper.converter.core.Converter;
+
+/**
+ * @author David Masclet
+ */
+@Path("/gisgraphy")
+@Produces("application/json; charset=utf-8")
+public class ConverterResourceGisgraphy extends AbstractConverterResource {
+
+	// all webservices
+	public static final String ADDRESS_PARAMETER = "address";
+	public static final String LAT_PARAMETER = "lat";
+	public static final String LONG_PARAMETER = "lng";
+	public static final String COUNTRY_PARAMETER = "country";
+	public static final String RADIUS_PARAMETER = "radius";
+	public static final String DEFAULT_FORMAT = "json";
+	public static final String FORMAT_PARAMETER = "format";
+	public static final String APIKEY_PARAMETER = "apikey";
+	// geocoding specid=fic
+	public static final String GEOCODING_LIMIT_PARAMETER = "limitnbresult";
+	// search
+	public static final String SEARCH_LIMIT_PARAMETER = "to";
+	public static final String SEARCH_QUERY_PARAMETER = "q";
+
+	public static final String REVERSE_QUERY_TYPE = "reverse";
+	public static final String GEOCODING_QUERY_TYPE = "geocoding";
+	public static final String AUTOCOMPLETE_QUERY_TYPE = "autocomplete";
+
+	private final String geocodingUrl;
+	private final String reverseGeocodingUrl;
+	private final String searchURL;
+	private final String apiKey;
+	private final Client jerseyClient;
+
+	public ConverterResourceGisgraphy(String geocodingUrl,
+			String reverseGeocodingUrl, String searchURL, String apiKey,
+			Client jerseyClient) {
+		this.geocodingUrl = geocodingUrl;
+		this.reverseGeocodingUrl = reverseGeocodingUrl;
+		this.searchURL = searchURL;
+		this.jerseyClient = jerseyClient;
+		this.apiKey = apiKey;
+	}
+
+	@GET
+	@Timed
+	public Response handle(@QueryParam("q") @DefaultValue("") String query,
+			@QueryParam("lat") @DefaultValue("") String lat,
+			@QueryParam("lng") @DefaultValue("") String lng,
```

Comment:
The GraphHopper API uses point instead of lat,lng, do you think it makes sense to point instead? I think it would make it easier to integrate this for our users?

Reply:
why not, but I propose to put default value to "" instead of false as in ConverterResourceNominatim#handle

Reply:
If autocomplete is true and so does reverse , do we 
1/ force reverse to false
2/ ignore autocomplete and consider it is a reverse query
3 / throws an exception
4 / other

Reply:
Good question. Maybe if autocomplete==true and reverse==true then we could do an autocomplete with location bias :) ?